client/js/lounge.js

```diff
@@ -428,16 +485,21 @@ $(function() {
 		}
 
 		if (prevMsgTime.toDateString() !== msgTime.toDateString()) {
-			prevMsg.append(render("date-marker", {msgDate: msgTime}));
+			var parent = prevMsg.parent();
+			if (parent.hasClass("condensed")) {
+				parent.append(render("date-marker", {msgDate: msgTime}));
+			} else {
+				prevMsg.append(render("date-marker", {msgDate: msgTime}));
```

Comment:
You can reduce the duplicate appends by simply doing `parent = prevMsg;` (with a better variable name) in the `if`.